### ISSUE TITLE
Question: How to make size of the loader fixed

### ISSUE THREAD

**Issue by bonlime** [NVIDIA/DALI#1203]
Hi,
I'll explain the issue on a simple example: I have 100 images and use a `batch_size` of 32. I read images using `FileReader` ops and feed them into my model using `for batch in loader:`. Sometimes I get 3 batches per epoch and sometimes 4. In native pytorch `Dataloader` there is a flag `drop_last` which makes sure that a not complete last batch is always dropped. Is there a way to make DALI behave the same way? 
such behavior is desirable for validation loader, where difference in number of batches makes val loss and val metrics differ during evaluation without any changes to the model.

**Comment by awolant** [comment]
Hi, thanks for the question.
Could you show me exact pipeline you are using and your code to read from it? 
In general, `FileReader` is not aware of the concept of an 'epoch'. It will read the samples from the dataset and transparently go back to the start.

**Comment by bonlime** [comment]
I use the basic pipeline, similar to the ones in the examples. `FileReader` is not aware of epoch but the pipelines knows when it runs out of samples and raises `StopIteration`. As far as I see pipe would yield batches till it has seen less than `ceil(pipe.epoch_size('Reader') / bs)` samples and then raise StopIteration. But during the next epoch Reader would start from the place it stopped last time and potentially could yield less number of batches. It's not an issue but I would like to somehow completely reset the reader before next epoch. (There is a `.reset()` method but it doesn't allow resetting not exhausted Reader)      
```python
class HybridPipe(Pipeline):
    def __init__(self, data_dir, filelist, sz, bs, num_threads, device_id):

        super(HybridPipe, self).__init__(bs, num_threads, device_id, seed=42)
        self.input = ops.FileReader(file_root=data_dir, file_list=filelist, random_shuffle=False)

        self.decode = ops.ImageDecoder(device="mixed", output_type=types.RGB)
        self.resize = ops.Resize(device='gpu', resize_shorter=int(sz*1.14))
        
        self.normalize = ops.CropMirrorNormalize(
            device="gpu",
            output_dtype=types.FLOAT,
            crop=(sz, sz),
            image_type=types.RGB,
            mean=[0.485 * 255,0.456 * 255,0.406 * 255],
            std=[0.229 * 255,0.224 * 255,0.225 * 255],
            output_layout=types.NCHW) 

    def define_graph(self):
        # Read images and labels
        images, labels = self.input(name="Reader")
        # Decode and augmentation
        images = self.decode(images)
        images = self.resize(images)
        images = self.normalize(images)

        return images, labels.gpu()

pipe = HybridPipe('dataset/images', filelist, sz, bs, num_workers, device_id, train)
pipe.build()
loader = PyTorchIterator(pipe, size=pipe.epoch_size('Reader'), auto_reset=True)
```

**Comment by awolant** [comment]
Ok, now I see.
I think you could use [`fill_last_batch`](https://github.com/NVIDIA/DALI/blob/master/dali/python/nvidia/dali/plugin/pytorch.py#L78) and [`last_batch_padded`](https://github.com/NVIDIA/DALI/blob/master/dali/python/nvidia/dali/plugin/pytorch.py#L89) parameters of `DALIGenericIterator` to control this behavior. They were added recently in #1005 and will be available soon in DALI 0.13.
If you want to use this features now, you are welcome to build DALI yourself or pick up one of the recent [nightly builds](https://docs.nvidia.com/deeplearning/sdk/dali-master-branch-user-guide/docs/installation.html#nightly-builds).

**Comment by bonlime** [comment]
Thx! That's exactly what I needed
